client/dashboard/sites/settings-repositories/index.tsx

```diff
@@ -1,33 +1,67 @@
 import { CodeDeploymentData, HostingFeatures } from '@automattic/api-core';
 import { siteBySlugQuery, codeDeploymentsQuery } from '@automattic/api-queries';
 import { useQuery, useSuspenseQuery } from '@tanstack/react-query';
+import { useRouter } from '@tanstack/react-router';
 import { Button, __experimentalText as Text } from '@wordpress/components';
 import { DataViews, filterSortAndPaginate } from '@wordpress/dataviews';
 import { __ } from '@wordpress/i18n';
 import { useState } from 'react';
-import { siteRoute } from '../../app/router/sites';
+import { siteDeploymentsListRoute, siteRoute } from '../../app/router/sites';
 import { DataViewsCard } from '../../components/dataviews-card';
 import PageLayout from '../../components/page-layout';
 import illustrationUrl from '../deployments/deployments-callout-illustration.svg';
 import ghIconUrl from '../deployments/gh-icon.svg';
+import { TriggerDeploymentModal } from '../deployments-list/trigger-deployment-modal';
 import HostingFeatureGatedWithCallout from '../hosting-feature-gated-with-callout';
 import SettingsPageHeader from '../settings-page-header';
 import { useRepositoryFields } from './dataviews/fields';
 import { DEFAULT_VIEW, DEFAULT_LAYOUTS } from './dataviews/views';
 import { DisconnectRepositoryModalContent } from './disconnect-repository-modal-content';
-import type { RenderModalProps, View } from '@wordpress/dataviews';
+import type { RenderModalProps, View, Action } from '@wordpress/dataviews';
 
 function RepositoriesList() {
+	const router = useRouter();
 	const { siteSlug } = siteRoute.useParams();
 	const { data: site } = useSuspenseQuery( siteBySlugQuery( siteSlug ) );
 	const [ view, setView ] = useState< View >( DEFAULT_VIEW );
+	const [ modalTriggerDeploymentDetails, setModalTriggerDeploymentDetails ] = useState( {
+		isOpen: false,
+		repositoryId: '',
+	} );
 
 	const { data: deployments = [], isLoading } = useQuery( codeDeploymentsQuery( site.ID ) );
 
 	const fields = useRepositoryFields();
 	const { data: filteredData, paginationInfo } = filterSortAndPaginate( deployments, view, fields );
 
-	const actions = [
+	const actions: Action< CodeDeploymentData >[] = [
+		{
+			id: 'trigger-manual-deployment',
+			label: __( 'Trigger manual deployment' ),
+			callback: ( items: CodeDeploymentData[] ) => {
+				setModalTriggerDeploymentDetails( {
+					isOpen: true,
+					repositoryId: items[ 0 ].id.toString(),
+				} );
+			},
+		},
+		{
+			id: 'configure-connection',
+			label: __( 'Configure connection' ),
+			callback: () => {},
+		},
+		{
+			id: 'see-deployment-runs',
+			label: __( 'See deployment runs' ),
+			callback: () => {
+				router.navigate( {
+					to: siteDeploymentsListRoute.fullPath,
+					params: {
+						siteSlug: siteSlug,
+					},
+				} );
+			},
+		},
 		{
 			id: 'delete',
```

Comment:
As discussed on [DOTDASH-376](https://linear.app/a8c/issue/DOTDASH-376), I think we should definitely differentiate the destructive action from the rest of the options.

Reply:
@sejas thanks for looking into this more! 🙌 

> In other parts of the V2 Dashboard we don't make that distinction. For example DNS records:

Thanks for highlighting this, I also tried to check but haven't found other instances in the Hosting Dashboard, nice catch! 👍 I also realized that we're using the deletion modal as a confirmation step before we actually remove the connection. Given these I think we're good to leave the delete option styled as is.